src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractCollectorBatchBuilder.java

```diff
@@ -179,16 +214,22 @@ protected void updateCollectorBatchWithOriginEntries() {
     protected boolean shouldProcessLine(String itemKey, ConcurStandardAccountingExtractDetailLine saeLine) {
         switch (saeLine.getPaymentCode()) {
             case ConcurConstants.PAYMENT_CODE_CASH :
+                return true;
             case ConcurConstants.PAYMENT_CODE_UNIVERSITY_BILLED_OR_PAID :
+                reportCorporateCardPayment(saeLine);
                 return true;
             case ConcurConstants.PAYMENT_CODE_PRE_PAID_OR_OTHER :
                 LOG.warn("Found a transaction with a Pre-Paid/Other payment code; it will not be processed. Item key: " + itemKey);
+                reportUnprocessedLine(saeLine, "The line has the Pre-Paid/Other (COPD) payment code");
                 return false;
             case ConcurConstants.PAYMENT_CODE_PSEUDO :
                 LOG.warn("Found a transaction with a Pseudo payment code; it will not be processed. Item key: " + itemKey);
+                reportBypassOfLineWithPseudoPaymentCode(saeLine);
+                reportUnprocessedLine(saeLine, "The line has the Pseudo (XXXX) payment code");
                 return false;
             default :
                 LOG.warn("Found a transaction with an unknown payment code; it will not be processed. Item key: " + itemKey);
+                reportUnprocessedLine(saeLine, "The line has an unrecognized payment code");

```

Comment:
The "item key" that I'm printing in the server log entries already contains the Report ID embedded in it. Also, the "reportUnprocessedLine" method is adding error items to the SAE report, and those items already have a field for the Report ID.